src/engine/src/scheduler/IScheduler.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <entt/entt.hpp>
+
+namespace ES::Engine {
+class Registry;
+}
+
+namespace ES::Engine::Scheduler {
+using USystem = std::function<void(Registry &)>;
+
+/**
+ * @brief Interface to be implemented for every schedulers
+ */
+class IScheduler {
```

Comment:
It's an abtract not an interface as you have an attribute in it

Reply:
yes, I should declare the interface without the registry variable and then using an abstract AScheduler for it